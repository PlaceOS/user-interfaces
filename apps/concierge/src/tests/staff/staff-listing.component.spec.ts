import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { MockComponent } from 'ng-mocks';
import { BehaviorSubject } from 'rxjs';
import { StaffDetailsComponent } from '../../app/staff/staff-details.component';
import { StaffListingComponent } from '../../app/staff/staff-listing.component';
import { StaffStateService } from '../../app/staff/staff-state.service';

describe('StaffListingComponent', () => {
    let spectator: Spectator<StaffListingComponent>;
    const createComponent = createComponentFactory({
        component: StaffListingComponent,
        providers: [
            {
                provide: StaffStateService,
                useValue: {
                    loading: new BehaviorSubject(false),
                    filtered_users: new BehaviorSubject([]),
                    startPolling: jest.fn(),
                    stopPolling: jest.fn(),
                },
            },
        ],
        declarations: [MockComponent(StaffDetailsComponent)],
        imports: [],
    });

    beforeEach(() => (spectator = createComponent()));

    it('should create component', () => {
        expect(spectator.component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(spectator.element).toMatchSnapshot();
    });

    it.todo('should allow scrolling to staff groupings');
    it.todo('should allow scrolling to staff groupings');
});
