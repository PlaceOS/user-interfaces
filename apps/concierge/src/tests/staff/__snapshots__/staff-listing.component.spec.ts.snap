// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaffListingComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="w-full p-2 flex items-center justify-center"
  >
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled active"
      letter=""
    >
       # 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       a 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       b 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       c 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       d 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       e 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       f 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       g 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       h 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       i 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       j 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       k 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       l 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       m 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       n 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       o 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       p 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       q 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       r 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       s 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       t 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       u 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       v 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       w 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       x 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       y 
    </div>
    <div
      class="capitalize h-6 w-6 flex items-center justify-center text-xs cursor-pointer disabled"
      letter=""
    >
       z 
    </div>
    
  </div>
  <div
    class="flex-1 overflow-auto w-full relative bg-gray-100"
    style="height: 50%;"
  >
    <div
      class="absolute inset-0 flex flex-col items-center justify-center"
    >
      <p>
        No matching staff members
      </p>
    </div>
    
  </div>
  
  
</div>
`;
